### PR TITLE
add `get_event_locked_tracking_insta_phase` function 

### PR DESCRIPTION
**Pull request by ttngu207**
+ Improve the json export function
+ Implemented `get_event_locked_tracking_insta_phase` as Dave suggested

### CODE REVIEW

File: pipeline/plot/behavior_plot.py [hunk 1]
```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
...
+            + experiment.ActionEvent
+        In the case that multiple of such "event_name" are found in a trial, the 1st one will be selected
+            (e.g. multiple "lick left", then the 1st "lick left" is selected)
+    :param d_name: any attribute name under the tracking.Tracking table - e.g. 'jaw_y', 'tongue_x', etc.
+    :return: list of instantaneous phase, in the same order of specified "trials"
+    """
+
+    trials = trials.proj()
+
+    tracking_fs = (tracking.TrackingDevice & tracking.Tracking & trials).fetch('sampling_rate')
```
Comment (ixcat) [inline pipeline/plot/behavior_plot.py]: can just do .fetch1 here

Comment (ttngu207) [reply pipeline/plot/behavior_plot.py]: I want to make sure we catch the cases where users specify trials from different recording type (hence different fs)

File: pipeline/plot/behavior_plot.py [hunk 2]
```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
...
+    jaw_types = [n for n in tracking.Tracking.JawTracking.heading.names if n not in tracking.Tracking.heading.names]
+
+    if d_name not in nose_types + tongue_types + jaw_types:
+        print(f'Unknown tracking type: {d_name}\nAvailable tracking types are: {nose_types + tongue_types + jaw_types}')
+        return
+
+    for trk_types, trk_tbl in zip((nose_types, tongue_types, jaw_types),
+                                  (tracking.Tracking.NoseTracking, tracking.Tracking.TongueTracking, tracking.Tracking.JawTracking)):
+        if d_name in trk_types:
+            d_tbl = trk_tbl
```
Comment (ixcat) [inline pipeline/plot/behavior_plot.py]: these tables are small, I'd just do:
`(tracking.Tracking.NoseTracking * tracking.Tracking.TongueTracking * tracking.Tracking.JawTracking).fetch('foo')` etc, where this is needed later and zap this block

File: pipeline/plot/behavior_plot.py [hunk 3]
```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
...
+            event_tbl = experiment.ActionEvent
+            eve_type_attr = 'action_event_type'
+            eve_time_attr = 'trial_event_time'
+        else:
+            print(f'Unknown event: {event}\nAvailable events are: {list(trial_event_types) + list(action_event_types)}')
+            return
+
+        tr_ids, trk_data, eve_times = trials.aggr(d_tbl, trk_data=d_name, keep_all_rows=True).aggr(
+            event_tbl & {eve_type_attr: event}, 'trk_data', event_time=f'min({eve_time_attr})', keep_all_rows=True).fetch(
+            'trial', 'trk_data', 'event_time', order_by='trial')
```
Comment (ixcat) [inline pipeline/plot/behavior_plot.py]: simiilarly the action/behavior events could be combined in the query e.g.:

```(dj.U('type') & experiment.TrialEventType.proj(type='trial_event_type')) + (dj.U('type') & experiment.ActionEventType.proj(type='action_event_type'))```

File: pipeline/plot/behavior_plot.py [hunk 4]
```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
...
+        eve_idx = eve_times.astype(float) * tracking_fs
+
+    else:
+        print('Unknown "event" argument!')
+        return
+
+    # ---- the computation part ----
+
+    # for trials with no jaw data (None), set to np.nan array
+    no_trk_trid = [idx for idx, jaw in enumerate(trk_data) if jaw is None]
```
Comment (ixcat) [inline pipeline/plot/behavior_plot.py]: can used masked arrays to avoid the nan filling/checking.

File: pipeline/plot/behavior_plot.py (hunk 1 above)
Comment (ixcat) [reply pipeline/plot/behavior_plot.py]: ahh.. missed the 'set' part. :)

**Pull request merged**